### PR TITLE
feat: Adding FORCE_SSL as feature flag in config.py

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY
Adding a feature flag for the new dynamic form being used for creating and editing databases. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N/A
### TESTING INSTRUCTIONS
go to config.py and look for FORCE_SSL

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by hughhhh — APPROVED** [review]

File: superset/config.py [hunk 1]
```diff
@@ -383,6 +383,7 @@ def _try_json_readsha(  # pylint: disable=unused-argument
     # for report with type 'report' still send with email and slack message with
     # screenshot and link
     "ALERTS_ATTACH_REPORTS": True,
+    "FORCE_SSL": False,
```
Comment (etr2460) [inline superset/config.py:386]: Can you add documentation about what this flag does?

Comment (AAfghahi) [reply superset/config.py:386]: Yeah! where should I add the documentation?

Comment (etr2460) [reply superset/config.py:386]: inline, similarly to the other feature flags would be great, thanks!

**Review by etr2460 — APPROVED** [review]
Thanks for the documentation! I think naming this `FORCE_DATABASE_CONNECTIONS_SSL` might be clearer since `FORCE_SSL` might imply that connecting to the webserver is forced with SSL, but up to you

**Comment by AAfghahi** [comment]
> Thanks for the documentation! I think naming this `FORCE_DATABASE_CONNECTIONS_SSL` might be clearer since `FORCE_SSL` might imply that connecting to the webserver is forced with SSL, but up to you

I like that, I am going to change it here and also in superset-ui

File: superset/config.py [hunk 2]
```diff
@@ -383,6 +383,9 @@ def _try_json_readsha(  # pylint: disable=unused-argument
     # for report with type 'report' still send with email and slack message with
     # screenshot and link
     "ALERTS_ATTACH_REPORTS": True,
+    # Enabling FORCE_SSL forces all database connections to be encrypted before
```
Comment (hughhhh) [inline superset/config.py:386]:
```suggestion
# Enabling FORCE_DATABASE_CONNECTIONS_SSL forces all database connections to be encrypted before
```

**Pull request merged**